Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -734,3 +763,165 @@ getStats driverId date = do
       { totalRidesOfDay = length rides,
         totalEarningsOfDay = sum . catMaybes $ rides <&> (.fare)
       }
+
+makeAlternatePhoneNumberKey :: Id SP.Person -> Text
+makeAlternatePhoneNumberKey id = "DriverAlternatePhoneNumber:PersonId-" <> id.getId
+
+makeAlternateNumberOtpKey :: Id SP.Person -> Text
+makeAlternateNumberOtpKey id = "DriverAlternateNumberOtp:PersonId-" <> id.getId
+
+makeAlternateNumberAttemptsKey :: Id SP.Person -> Text
+makeAlternateNumberAttemptsKey id = "DriverAlternateNumberAttempts:PersonId-" <> id.getId
+
+cacheAlternateNumberInfo :: (CacheFlow m r) => Id SP.Person -> Text -> Text -> Int -> m ()
+cacheAlternateNumberInfo personId phoneNumber otp attempts = do
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+  let alternatePhoneNumberKey = makeAlternatePhoneNumberKey personId
+  let alternateNumberOtpKey = makeAlternateNumberOtpKey personId
+  let alternateNumberAttemptsKey = makeAlternateNumberAttemptsKey personId
+
+  Redis.setExp alternatePhoneNumberKey phoneNumber expTime
+  Redis.setExp alternateNumberOtpKey otp expTime
+  Redis.setExp alternateNumberAttemptsKey attempts expTime
+
+validateAlternateNumber ::
+  ( EsqDBFlow m r,
+    EncFlow m r,
+    Redis.HedisFlow m r,
+    HasCacheConfig r,
+    CoreMetrics m,
+    HasFlowEnv m r ["apiRateLimitOptions" ::: APIRateLimitOptions, "smsCfg" ::: SmsConfig],
+    HasFlowEnv m r '["altenateOtpSmsTemplate" ::: Text],
+    HasAternateNumberAttempts r
+  ) =>
+  Id SP.Person ->
+  DriverAlternateNumberReq ->
+  m APISuccess
+validateAlternateNumber personId phoneNumber = do
+  Redis.whenWithLockRedis (makeAlternatePhoneNumberKey personId) 60 $ do
+    mobileNumberHash <- getDbHash phoneNumber.alternateNumber
+    person <- QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+    duplicateCheck (QPerson.findByMobileNumber phoneNumber.mobileCountryCode mobileNumberHash) "Person with this mobile number already exists"
+    smsCfg <- asks (.smsCfg)
+    let useFakeOtpM = useFakeSms smsCfg
+    let fakeOtp = show <$> useFakeOtpM
```

Comment:
These type of lines can be combined, we can use only one variable(applies everywhere)